### PR TITLE
[FIX]Fix oom nodes.

### PR DESCRIPTION
**Pull request by lazavikmaria**
Was created variable "assign_to_space00_07_nodes" ==> All instances have label  
assign_to_space00_07_nodes = allow_any_pods by default value (false).
 If we need that only critical pods assign to only critical nodes ( nodegroup 5, nodegroup 7 == space00, space07) switch on value  (true)  == >
assign_to_space00_07_nodes = allow_only_critical_pods. 

!!! IMPORTANT !!!
After apply all instances dev + mainnet will restart.

### CODE REVIEW

File: modules/eks_nodegroup/launch_templates_ondemand.tf [hunk 1]
```diff
@@ -47,6 +47,7 @@ resource "aws_launch_template" "lt_ondemand" {
         "Name"          = "${module.generator.prefix}-${local.get_nodegroup_postfix}-ondemand",
         "Type"          = "ondemand"
         "nodeGroupName" = local.get_nodegroup_postfix
+        "assign_pods_to_key_nodes" = var.assign_pods_to_key_nodes
```
Comment (witnsby) [inline modules/eks_nodegroup/launch_templates_ondemand.tf:50]: could I ask you to make fmt code before apply it?
terrafrom fmt in the folder

File: modules/eks_nodegroup/launch_templates_spot.tf [hunk 2]
```diff
@@ -47,6 +47,7 @@ resource "aws_launch_template" "lt_spot" {
         "Name"          = "${module.generator.prefix}-${local.get_nodegroup_postfix}-spot",
         "Type"          = "spot",
         "nodeGroupName" = local.get_nodegroup_postfix
+        "assign_pods_to_key_nodes" = var.assign_pods_to_key_nodes
```
Comment (witnsby) [inline modules/eks_nodegroup/launch_templates_spot.tf:50]: could I ask you to make fmt code before apply it?
terrafrom fmt in the folder

File: modules/eks_nodegroup/locals.tf [hunk 3]
```diff
@@ -24,6 +24,7 @@ locals {
       "lifecycle"     = var.is_spot_instance ? "spot" : "ondemand"
       "nodeGroupName" = local.get_nodegroup_postfix
       "environment"   = local.get_environment
+      "assign_pods_to_key_nodes" = var.assign_pods_to_key_nodes == true ? "allow_only_critical_pods" : "allow_any_pods"
```
Comment (witnsby) [inline modules/eks_nodegroup/locals.tf:27]: could I ask you to make fmt code before apply it?
terrafrom fmt in the folder

**Review by witnsby — APPROVED** [review]

File: modules/eks_nodegroup/launch_templates_ondemand.tf (hunk 1 above)
Comment (lazavikmaria) [reply modules/eks_nodegroup/launch_templates_ondemand.tf:50]: Was applied fmt format.

File: modules/eks_nodegroup/launch_templates_spot.tf (hunk 2 above)
Comment (lazavikmaria) [reply modules/eks_nodegroup/launch_templates_spot.tf:50]: Was applied fmt format.

File: modules/eks_nodegroup/locals.tf (hunk 3 above)
Comment (lazavikmaria) [reply modules/eks_nodegroup/locals.tf:27]: Was applied fmt format.

**Review by witnsby — DISMISSED** [review]

**Review by ArseniiPetrovich — CHANGES_REQUESTED** [review]
Well, I have a concern. 
It is not clear on what the "key nodes" are. It is not described anywhere in the docs, it is not a part of the projects. I think that the variables selection should be changed and explicitly show the nodes that we want/don't want to assign to

**Review by ArseniiPetrovich — APPROVED** [review]

**Pull request merged**